src/ooni/ooni_test_impl.hpp

```diff
@@ -69,10 +70,40 @@ class OoniTestImpl : public mk::NetTest {
             });
     }
 
-    void geoip_lookup() {
+    void geoip_lookup(Callback<> cb) {
         probe_ip = "127.0.0.1";
         probe_asn = "AS0";
         probe_cc = "ZZ";
+        ip_lookup([=](Error err, std::string ip) {
+            if (!err) {
```

Comment:
Maybe it would be useful to have at least some logging be printed when the geoip lookup fails.